pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -187,18 +189,25 @@ func Execute(configFile io.Reader) {
 		log.Fatal(err)
 	}
 
-	// signal has been received. we will attempt to stop our http server
-	// using a timeout slightly lower than the default k8s grace period
-	// of 30 seconds.
-	ctx, cancel := context.WithTimeout(ctx, 29*time.Second)
-	defer cancel()
+	dcontext.GetLogger(ctx).Infof("setting service to unready")
+	setReady(false)
+	time.Sleep(35 * time.Second)
```

Comment:
we still want this to be 55s right?  i.e. we want to stick around as long as possible closing existing connections, right up until the termination grace period(which is 55s) expires if necessary

Reply:
We want this to be readinessPeriod*Failurethreshold (10x3+fudge=35)... this allows the readiness to get propagated to the api/load-balancers/etc. After this... The rest of the shutdown proceeds. I removed the timeout on the HTTP server shutdown. By removing the http server timeout all the connections will gracefully terminate. The pod has a hard stop of 55 seconds already with the terminationGracePeriodSeconds being set to 55.